### PR TITLE
Add short doc explaining the configuration module

### PR DESCRIPTION
**Pull request by tanertopal**
Feel free to modify and/or merge

### CODE REVIEW

File: xain/CONFIG.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Configuration
+Currently configuration happens two fold.
+
+## `config.py`
+On the initial run of the library the `xain.config.init_config()` function
+will be executed and initialize a `config.cfg` file in the projects root directory.
```
Comment (r-marques) [inline xain/CONFIG.md]: should this be `config.py`?

File: xain/CONFIG.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Configuration
+Currently configuration happens two fold.
+
+## `config.py`
+On the initial run of the library the `xain.config.init_config()` function
+will be executed and initialize a `config.cfg` file in the projects root directory. 
+If the user wants to run benchmarks and wants results to be uploaded to S3 he will need to replace the
+`ACCESSIBLE_S3_BUCKET_FOR_RESULTS_TO_BE_UPLOADED` string in the `config.cfg` with an actual S3 bucket name. The bucket
```
Comment (r-marques) [inline xain/CONFIG.md]: `config.py`?

File: xain/CONFIG.md (hunk 1 above)
Comment (tanertopal) [reply xain/CONFIG.md]: Nope. the `config.py` resides in `xain/config.py`.  In the top level `init.py` we call xain.config.init_config() if no `config.cfg` is yet present in the top level directory and create it in that case.

File: xain/CONFIG.md (hunk 2 above)
Comment (tanertopal) [reply xain/CONFIG.md]: Same as above :)

**Pull request merged**